contracts/src/v0.8/ccip/libraries/Internal.sol

```diff
@@ -75,21 +75,25 @@ library Internal {
     uint256 proofFlagBits;
   }
 
-  /// @dev Any2EVMRampMessage struct has 10 fields, including 3 variable unnested arrays (data, receiver and tokenAmounts).
+  /// @dev Any2EVMRampMessage struct has 10 fields, including 3 variable unnested arrays (sender, data and tokenAmounts).
   /// Each variable array takes 1 more slot to store its length.
   /// When abi encoded, excluding array contents,
   /// Any2EVMMessage takes up a fixed number of 13 slots, 32 bytes each.
+  /// Assume 1 slot for sender
   /// For structs that contain arrays, 1 more slot is added to the front, reaching a total of 14.
   /// The fixed bytes does not cover struct data (this is represented by MESSAGE_FIXED_BYTES_PER_TOKEN)
-  uint256 public constant MESSAGE_FIXED_BYTES = 32 * 14;
+  uint256 public constant MESSAGE_FIXED_BYTES = 32 * 15;
 
-  /// @dev Each token transfer adds 1 RampTokenAmount
-  /// RampTokenAmount has 5 fields, 2 of which are bytes type, 1 Address, 1 uint256 and 1 uint32.
-  /// Each bytes type takes 1 slot for length, 1 slot for data and 1 slot for the offset.
-  /// address
-  /// uint256 amount takes 1 slot.
-  /// uint32 destGasAmount takes 1 slot.
-  uint256 public constant MESSAGE_FIXED_BYTES_PER_TOKEN = 32 * ((2 * 3) + 3);
+  /// 0x20
+  /// destTokenAddress
+  /// destGasAmount
+  /// amount
+  /// sourcePoolAddress_offset
+  /// extraData_offset
+  /// sourcePoolAddress_length
+  /// sourcePoolAddress_content // assume 1 slot
+  /// extraData_length // contents billed separately
```

Comment:
nit: I think if we follow the struct layout the order would be:
```
/// 0x20
/// sourcePoolAddress_offset
/// destTokenAddress
/// destGasAmount
/// extraData_offset
/// amount
/// sourcePoolAddress_length
/// sourcePoolAddress_content // assume 1 slot
/// extraData_length // contents billed separately
```

Reply:
I wrote it in terms of how we group the slots and use 4, 3 and 2 in the following calculation. Let me know

Reply:
Making the correction